src/utils/kmlUtils.js

```diff
@@ -399,6 +401,8 @@ export function getEditableFeatureFromKmlFeature(kmlFeature, availableIconSets)
     const iconSize = iconStyle ? getIconSize(iconStyle) : null
     const fillColor = getFillColor(style, kmlFeature.getGeometry().getType(), iconArgs)
 
+    const textOffset =
+        iconSize && !kmlLayer.isLegacy() ? getTextOffset(textScale, iconSize, iconStyle) : [0, 0]
```

Comment:
Would make more sense to have something like this
```javascript
let textOffset = [0, 0]
if (!kmlLayer.isLegacy() && iconSize && featureType === EditableFeatureTypes.MARKER) {
  textOffset = getTextOffset(...)
}
```
Instead of checking the type in the EditableFeature, this would keep the last one very generic.